game/src/core/components/DrawComponent.java

```diff
@@ -206,4 +231,45 @@ public boolean isCurrentAnimationLooping() {
     public boolean isCurrentAnimationFinished() {
         return currentAnimation.isFinished();
     }
+
+    public boolean reachedFrametimeLimit() {
+        return frameCounter >= currentFrametime;
+    }
+
+    // checks the status of animations in the animationQueue and selects the next animation by
+    // priority
+    public void setNextAnimation() {
+
+        if (animationQueue.size() > 0) {
+            Animation highestPrio = null;
+
+            // iterate through animationQueue
+            for (Map.Entry<IPath[], Integer> animationArr : animationQueue.entrySet()) {
+                // subtract 1 from every frametimer, if value below zero, remove animation from
+                // queue
+                animationArr.setValue(animationArr.getValue() - 1);
+                if (animationArr.getValue() < 0) {
+                    animationQueue.remove(animationArr.getKey());
+                    break;
+                }
+
+                // if animation has frametime left, check if it's the highest priority
+                // then generate the first valid Animation from that array
+                for (IPath animationPath : animationArr.getKey()) {
+                    if (highestPrio == null
+                            || highestPrio.getPriority() < animationPath.priority()) {
+                        highestPrio = animationMap.get(animationPath.pathString());
+                        if (highestPrio != null) {
+                            currentAnimation = highestPrio;
+                            break;
+                        } else
+                            LOGGER.warning(
+                                    "Animation "
+                                            + animationPath
+                                            + " can not be set, because the given Animation could not be found.");
```

Comment:
Die Log nachricht verwirrt mich, ist das nicht eher "Es wurde keine Animation gesetzt"?  aka die queue ist leer?

Reply:
ist an der stelle in setNextAnimation() rausgeflogen, da Animation nun über DrawComponent#currentAnimation(IPath) gesetzt wird, und der Logger da schon drin ist